macros/edr/tests/test_exposure_schema_validity.sql

```diff
@@ -0,0 +1,64 @@
+{% test exposure_schema_validity(model, exposures, node, columns) %}
+    {%- if not execute -%}
+        {%- do return(none) -%}
+    {%- endif -%}
+
+    {# Parameters used only for dependency injection in integration tests #}
+    {%- set node = node | default(context['model']['attached_node']) -%}
+    {%- set exposures = ( exposures | default(graph.exposures) ).values() -%}
+    {%- set columns = columns | default(adapter.get_columns_in_relation(model))  -%}
+
+    {%- set model_relation = elementary.get_model_relation_for_test(model, context["model"]) -%}
+    {%- set full_table_name = elementary.relation_to_full_name(model_relation) -%}
+    {{- elementary.test_log('start', full_table_name, 'exposure validation') -}}
+
+    {%- set matching_exposures = [] -%}
+
+    {%- for exposure in exposures -%}
+        {%- if node in exposure.depends_on.nodes and exposure['meta'] | default(none) is not none -%}
+            {%- do matching_exposures.append(exposure) -%}
```

Comment:
I'd expect the `matching_exposures` to only be exposures that are referencing the tested model in `depends_on`. This would also save a lot of unnecessary rendering when there are a lot of exposures.

Reply:
That's what this loop does